### DISCUSSION TITLE
create a script to download all models at once (localmodels.sh for example)

### DISCUSSION THREAD

**Discussion by koinkoin-project** [Woolverine94/biniou#1] [Ideas]
The idea would be to give the biniou user the means to download all available models at once, without having to download them one by one. This could be done via a script, which you could call localmodels.sh or any other name that suits you.

Advantages:

Saves time for users who need multiple templates.
Simplified download process.
Reduced number of interactions required to obtain all templates.

Thks

**Comment by Woolverine94** [comment]
Hello @koinkoin-project , 

Thanks for the suggestion. 

I will think about a way of doing it efficiently, maybe by creating an "offline installer", which will prepare biniou for an offline use. 

Downloading all defaults models will require 100GB+ and it would be (at least) 3 times this size for all models listed in the biniou interface. 

Please also note that the models folder of biniou is only a redirection of (most) content of your .cache/huggingface directory and only a more convenient way to manage the cache model, but is not a really clever way to manage models.

**Comment by koinkoin-project** [comment]
:+1:

**Comment by koinkoin-project** [comment]
Yes, it would be wise to inform users via the console that the download process will be lengthy depending on their network connection speed. It could take up to a few hours with a fiber connection and significantly longer, especially with an ADSL connection. Additionally, it would be advisable to remind them to allocate at least 500 gigabytes of disk space to ensure they have enough available storage. While users who understand the purpose of this script and initiate it should have an idea of what to expect, displaying a brief reminder certainly wouldn't hurt.

**Comment by koinkoin-project** [comment]
> Please also note that the models folder of biniou is only a redirection of (most) content of your .cache/huggingface directory and only a more convenient way to manage the cache model, but is not a really clever way to manage models.

Perhaps it's not the best solution, but have you considered doing the exact opposite? In other words, retrieving all the models directly from the .cache/huggingface directory. In this case, the script would be much simpler. However, I want to emphasize that my knowledge of generative AI, Biniou, Gradio, and other subjects is somewhat limited. I apologize in advance if this suggestion is not relevant.

**Reply by Woolverine94** [reply]
In fact, it's a very good idea, and probably the simplest way to have a unique consistent folder for all models : some default path models seems hardcoded in the libraries, and thus could not be written simply outside of the cache directory.
I will give it a try. 
Thanks for the proposal.
